sos/report/__init__.py

```diff
@@ -254,7 +253,7 @@ def add_parser_options(cls, parser):
                             dest="compression_type",
                             default='auto',
                             help="compression technology to use [auto, "
-                                 "gzip, bzip2, xz] (default=auto)")
+                                 "gzip, xz] (default=auto)")
```

Comment:
We should add a `choices` parameter here to `report` like we have in `collect`, as otherwise users can still specify `bzip2`, but get an `xz` compressed archive.

```
$ sudo ./bin/sos report --batch -z bzip2

sosreport (version 3.9)

This command will collect system configuration and diagnostic
information from this Fedora system.
[...]
Your sosreport has been generated and saved in:
  /var/tmp/sosreport-terra-2020-05-08-mitmxbv.tar.xz
```

vs.

```
$ sudo ./bin/sos collect --batch -z bzip2
usage: sos collect [options]
sos <component> [options]

collect: error: argument -z/--compression-type: invalid choice: 'bzip2' (choose from 'auto', 'gzip', 'xz')
```